LRRServer/lawreadingrobot/main.py

```diff
@@ -0,0 +1,64 @@
+import traceback
+from typing import Any, Callable, Generator, List
+
+from fastapi import Depends, FastAPI, Request
+from fastapi.middleware.cors import CORSMiddleware
+from sqlalchemy.orm import Session
+
+from . import crud, models, schemas
+from .database import SessionLocal, engine
+
+models.Base.metadata.create_all(bind=engine)
+
+app = FastAPI()
+
+origins = [
+    "http://localhost",
+    "http://localhost:5173",
+]
+
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=origins,
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+
+async def catch_exceptions_middleware(request: Request, call_next: Callable[..., Any]) -> None:
+    try:
+        return await call_next(request)
+    except Exception as e:
+        print(traceback.format_exc())
+        raise e
+
+
+app.middleware('http')(catch_exceptions_middleware)
```

Comment:
Can you separate the function definitions from the main code path? Maybe with an explicit setup function:
```
def setup():
   """Creates the database if necessary, set up cross origin resource sharing and begin serving with FastAPI."""
   models.Base.metadata.create_all(bind=engine)
   app = FastAPI()
   # etc...
   
if __name__ = '__main__':
  setup()
```

Reply:
I'd say this is a nice-to-have, but not super necessary. If we don't get at this now, we should probably just note it for clean-up later.

Reply:
the main module is meant to be run this way with fast api. if you're using `uvicorn` or `gunicorn` to run the application the `__name__` of the module will be `lawreadingrobot.main` in this case.